### DISCUSSION TITLE
"-nan" in the fitting results of quartic IFCs

### DISCUSSION THREAD

**Discussion by JoeyyHuang** [ttadano/alamode#68] [Q&A]
Dear T. Tadano,
I am a new user of ALAMODE with latest version 1.3.0. Following the tutorial of Si example, I can get the Harmonic IFCs and phonon dispersion smoothly, then I try to fit the quartic IFCs for scph calculation.  However, after set CV=19 for cross-validation automatically, I get a lot of "-nan" in the cvscore file as follow:

 Algorithm : Coordinate descent
 Linear model : Elastic-net
 L1_RATIO = 0.8
 ENET_DNORM =               1
 STANDARDIZE = 1
 CONV_TOL =           1e-13
 95-fold cross-validation scores
 L1 ALPHA, Fitting error (mean, std), Validation error (mean, std)
    0.000780246           -nan           -nan           -nan           -nan
    0.000591877           -nan           -nan           -nan           -nan
    0.000448985           -nan           -nan           -nan           -nan
     0.00034059           -nan           -nan           -nan           -nan
    0.000258364           -nan           -nan           -nan           -nan
    0.000195989           -nan           -nan           -nan           -nan
    0.000148673           -nan           -nan           -nan           -nan
     0.00011278           -nan           -nan           -nan           -nan
    8.55523e-05           -nan           -nan           -nan           -nan
     6.4898e-05           -nan           -nan           -nan           -nan
    4.92302e-05           -nan           -nan           -nan           -nan
    3.73449e-05           -nan           -nan           -nan           -nan
     2.8329e-05           -nan           -nan           -nan           -nan
    2.14898e-05           -nan           -nan           -nan           -nan
    1.63016e-05           -nan           -nan           -nan           -nan
    1.23661e-05           -nan           -nan           -nan           -nan

So I can't get the best  L1 ALPHA value with the minimum value of the validation score.

I check the solution_path files, and find that for every L1 ALPHA, there is one "-nan" in the coefficients as follow:

 L1 ALPHA, coefficients
    0.000780246             -0             -0             -0             -0   -0.000168293   -0.000406501              0             -0             -0             -0             -0              0             -0              0             -0              0              0              0              0             -0             -0              0             -0              0              0              0             -0              0            nan              0             -0              0              0             -0             -0             -0              0             -0              0              0              0             -0              0             -0              0              0              0              0             -0             -0              0              0              0              0              0             -0             -0              0              0             -0              0             -0              0             -0             -0              0              0             -0             -0              0              0             -0             -0             -0              0              0             -0             -0              0             -0             -0             -0             -0              0              0              0
    0.000591877             -0             -0             -0             -0   -0.000320923   -0.000559131              0             -0             -0             -0             -0              0             -0              0             -0              0              0              0              0             -0             -0              0             -0              0              0              0             -0              0            nan              0             -0              0              0             -0             -0             -0              0             -0              0              0              0             -0              0             -0              0              0              0              0             -0             -0              0              0              0              0              0             -0             -0              0              0             -0              0             -0              0             -0             -0              0              0             -0             -0              0              0             -0             -0             -0              0              0             -0             -0              0             -0             -0             -0             -0              0              0              0
    0.000448985             -0             -0             -0             -0   -0.000436704   -0.000674913              0             -0             -0             -0             -0              0             -0              0             -0              0              0              0              0             -0             -0              0             -0              0              0              0             -0              0            nan              0             -0              0              0             -0             -0             -0              0             -0              0              0              0             -0              0             -0              0              0              0              0             -0             -0              0              0              0              0              0             -0             -0              0              0             -0              0             -0              0             -0             -0              0              0             -0             -0              0              0             -0             -0             -0              0              0             -0             -0              0             -0             -0             -0             -0              0              0              0
     0.00034059             -0             -0             -0             -0   -0.000524534   -0.000762742              0             -0             -0             -0             -0              0             -0              0             -0              0              0              0              0             -0             -0              0             -0              0              0              0             -0              0            nan              0             -0              0              0             -0             -0             -0              0             -0              0              0              0             -0              0             -0              0              0              0              0             -0             -0              0              0              0              0              0             -0             -0              0              0             -0              0             -0              0             -0             -0              0              0             -0             -0              0              0             -0             -0             -0              0              0             -0             -0              0

Here is part of the input file:

&general
   PREFIX = alamode
   MODE = optimize
   NAT = 64; NKD = 1
   KD = Si
/
&interaction
   NORDER = 3  # 1: harmonic, 2: cubic, ..
   NBODY = 2 3 3
/
&optimize
   LMODEL = enet
   DFSET = DFSET_ANHARM4
   NDATA = 95
   FC2XML = ../alm-2/alamode.xml
   CONV_TOL = 1.0e-13
   MAXITER = 50000
   L1_RATIO = 0.8
   CV = 19
   SOLUTION_PATH = 1
/ 
&celL
   20.53404573
   1.0 0.0 0.0 # a1
   0.0 1.0 0.0 # a2
   0.0 0.0 1.0 # a3
/
&cutoff 
   *-* None 10 6
/

The number of displacement-force training datasets(95) is larger than the number of Total Number of Free Parameters(86). I also have changed the CV value but the problem remains. Have you some ideas why this happen?

**Comment by ttadano** [comment]
Hi,

Could you try the following?

1. Use `L1_RATIO = 1` (LASSO)
2. Use a smaller value for `CV`, say `CV = 5`.

**Reply by JoeyyHuang** [reply]
Dear Dr. Tadano,
Thank you for the suggestion. change the L1_RATIO and CV values do not work. But the problem is solved by enlarging the cutoff radius for quartic terms(from - None 10 6 to - None None None). Maybe no need to use so large cutoff radius, anyway, it works now.
